Standards/CCSMB-11.md

```diff
@@ -0,0 +1,99 @@
+
+| Field             | Type                | Description                                                                                                                       |
+|---------------------------|---------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Name                      | str                         | A human-readable name of the application.                                                                                                                                                                                    |
+| Type                      | str ("Application")         | The type of the entry. Compliant software SHOULD ignore the whole entry if this is set to an unknown value.                                                                                                                  |
+| Exec                      | str                         | The path to the file to execute, including arguments. The full path MAY be omitted if it is present in `shell.getPath`.                                                                                                      |
+| Terminal                  | bool                        | Whether to run the application in a terminal or not.                                                                                                                                                                         |
+| CompliantWindowManagement | bool                        | Whether the application manages its own windows through a system-specific API. If set to `false`, compliant specifications SHOULD define `ShowIn` (see below). If `Terminal` is set to `true`, this is implied to be `true`. |
+
+Compliant application specifications MAY define the following additional fields:
+
+| Field       | Type                        | Description                                                                                                                                                                                                             |
+|-------------|-----------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Icon        | str-indexed array of str    | An array, where the keys are sizes in pixels (16x16 and 32x32 are common) and the values are paths to a square image representing the application. The file MAY be a BIMG file; other formats are permitted but discouraged.                                                                                      |
```

Comment:
keys of text icons should be in the format `text_`_width_`x`_height_ and bitmap icons should be in the format `bitmap_`_width_`x`_height_ (idk what bitmap image format it should use, maybe xpm?)

Reply:
> keys of text icons

Could you elaborate on what you mean by text icons?

Reply:
stuff like bimg

Reply:
> keys of text icons should be in the format `text_`_width_`x`_height_ and bitmap icons should be in the format `bitmap_`_width_`x`

I think a nested array setup would be better than the current one:

```lua
Icons = {
  { Type = "Vector", Width = "16", Height = "16", Path = "/path/to/icon16x16.xpm" };
  { Type = "Bitmap", Width = "16", Height = "16", Path = "/path/to/icon16x16.bimg" };
}
```

Thoughts?

> (idk what bitmap image format it should use, maybe xpm?)

Not sure. I'm open to input on this part.

Reply:
why is the width and height a string???

Reply:
That was a mistake in this example. The actual specification and the in-source examples both use integers.